### PR TITLE
docs: update iOS performance findings

### PR DESCRIPTION
**Pull request by rebello95**
Updating based on https://github.com/lyft/envoy-mobile/issues/128#issuecomment-516260951.

### CODE REVIEW

File: docs/root/development/performance/device_conditions.rst [hunk 1]
```diff
@@ -30,7 +33,45 @@ Experimentation method
...
+a device with debugging symbols:
+
+``bazel build ios_dist --config=ios --ios_multi_cpus=arm64 --copt=-ggdb3``
+
+In the active scheme of the app's ``Environment Variables``, set ``CFNETWORK_DIAGNOSTICS=3``
+to enable more verbose ``CFNetwork`` logs. Additionally, set Envoy's logs to ``trace``.
+
+Within the app, we performed the following:
+
+- Made single requests to a simple Python server from which we could see when the client connected, disconnected, and made requests
```
Comment (buildbreaker) [inline docs/root/development/performance/device_conditions.rst]: Are you able to link the python server code?

File: docs/root/development/performance/device_conditions.rst [hunk 2]
```diff
@@ -30,7 +33,45 @@ Experimentation method
 Modified versions of the "hello world" example apps were used to run the following experiments.
 See the iOS/Android sections below for instructions on building the examples.
 
-Throughout all of the following steps, we tested to make sure that network requests succeeded
+iOS
```
Comment (buildbreaker) [inline docs/root/development/performance/device_conditions.rst]: I feel like these are general observations with some actions. Could we instead write up very specific steps on how you were able to reproduce the experimentation.

Something like:
```
1. Start the python service
2. Switched between WiFi to cellular
3. Wait for logs to appear
...
``

This way a reader can precisely reproduce the experimentation steps and read more about the observations you noticed in the analysis

File: docs/root/development/performance/device_conditions.rst (hunk 1 above)
Comment (rebello95) [reply docs/root/development/performance/device_conditions.rst]: Sure

File: docs/root/development/performance/device_conditions.rst (hunk 2 above)
Comment (rebello95) [reply docs/root/development/performance/device_conditions.rst]: Updated

**Pull request merged**